---
title: "Workshop Rmd assignment"
author: "Giuseppe Tarantino & Eggtart Chen"
date: "2021-06-28"
output:
  html_document: 
    code_folding: hide
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
    keep_md: yes
---

<img src="./eggtart.jpg" width="15%" />


# Set-up libraries and data
## Load libraries

```r
library(tidyverse)
```

```
## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──
```

```
## ✓ ggplot2 3.3.3     ✓ purrr   0.3.4
## ✓ tibble  3.1.0     ✓ dplyr   1.0.5
## ✓ tidyr   1.1.3     ✓ stringr 1.4.0
## ✓ readr   1.4.0     ✓ forcats 0.5.0
```

```
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
```

```r
library(pheatmap)
library(circlize)
```

```
## ========================================
## circlize version 0.4.11
## CRAN page: https://cran.r-project.org/package=circlize
## Github page: https://github.com/jokergoo/circlize
## Documentation: https://jokergoo.github.io/circlize_book/book/
## 
## If you use it in published research, please cite:
## Gu, Z. circlize implements and enhances circular visualization
##   in R. Bioinformatics 2014.
## 
## This message can be suppressed by:
##   suppressPackageStartupMessages(library(circlize))
## ========================================
```

```r
library(ComplexHeatmap) #I prefer this package for heatmap visualization
```

```
## Loading required package: grid
```

```
## ========================================
## ComplexHeatmap version 2.4.3
## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
## Github page: https://github.com/jokergoo/ComplexHeatmap
## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference
## 
## If you use it in published research, please cite:
## Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
##   genomic data. Bioinformatics 2016.
## 
## This message can be suppressed by:
##   suppressPackageStartupMessages(library(ComplexHeatmap))
## ========================================
```

## Load data


```r
load("data/Rmarkdown_data.Rdata")
```

# Top 20 significant genes

## Get names of top 20 genes

```r
top20_sigOE_genes <- res_tableOE_tb %>% 
  arrange(padj) %>% 	#Arrange rows by padj values
  pull(gene) %>% 		#Extract character vector of ordered genes
  head(n=20)

top20_sigOE_genes
```

```
##  [1] "MOV10"    "H1F0"     "HSPA6"    "HIST1H1C" "TXNIP"    "NEAT1"   
##  [7] "KLF10"    "INSIG1"   "NR1D1"    "WDFY1"    "HSPA1A"   "HMGCS1"  
## [13] "LAMC1"    "HSPA1B"   "TMCO1"    "ADAMTS1"  "MRC2"     "C1orf95" 
## [19] "ZFP36L1"  "COL2A1"
```
## normalized counts for top 20 significant genes

```r
top20_sigOE_norm <- normalized_counts %>%
  filter(gene %in% top20_sigOE_genes)
```

## Gathering the columns to have normalized counts to a single column

```r
gathered_top20_sigOE <- top20_sigOE_norm %>%
  gather(colnames(top20_sigOE_norm)[2:9], key = "samplename", value = "normalized_counts")
gathered_top20_sigOE <- inner_join(mov10_meta, gathered_top20_sigOE)
```

```
## Joining, by = "samplename"
```

## plot using ggplot2

```r
ggplot(gathered_top20_sigOE) +
  geom_point(aes(x = gene, y = normalized_counts, color = sampletype)) +
  scale_y_log10() +
  xlab("Genes") +
  ylab("log10 Normalized Counts") +
  ggtitle("Top 20 Significant DE Genes") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(plot.title = element_text(hjust = 0.5))
```

![](Rscript_files/figure-html/plot_genes-1.png)<!-- -->

We expected to see this difference in MOV10, it is interesting to see how much HSPA6 is modulated by MOV10

# Create a heatmap of the differentially expressed genes

## Extract normalized expression for significant genes from the OE and control samples (2:4 and 7:9)

```r
res_tableOE_tb_sig <- res_tableOE_tb %>%
  filter(padj < 0.05)
```

## Return the normalized counts for the significant DE genes{.tabset .tabset-fade .tabset-pills}

```r
norm_OEsig <- normalized_counts %>% 
  filter(gene %in% res_tableOE_tb_sig$gene) 

meta <- mov10_meta %>%
column_to_rownames("samplename") %>%
data.frame()
```

### Run pheatmap using the metadata data frame for the annotation

```r
pheatmap(norm_OEsig[2:9], 
         cluster_rows = T, 
         show_rownames = F,
         annotation = meta, 
         border_color = NA, 
         fontsize = 10, 
         scale = "row", 
         fontsize_row = 10, 
         height = 20)
```

![](Rscript_files/figure-html/heatmap-1.png)<!-- -->

### Using ComplexHeatmap


```r
column_ha = HeatmapAnnotation(sampletype = meta$sampletype, MOVexpr=meta$MOVexpr)
scaled_mat<-scale(t(norm_OEsig[2:9]))
Heatmap(t(scaled_mat),col=colorRamp2(c(2, 0, -2), c("blue", "white", "red")), top_annotation = column_ha)
```

![](Rscript_files/figure-html/heatmap_v2-1.png)<!-- -->

## Session Information

```r
sessionInfo()
```

```
## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] ComplexHeatmap_2.4.3 circlize_0.4.11      pheatmap_1.0.12     
##  [4] forcats_0.5.0        stringr_1.4.0        dplyr_1.0.5         
##  [7] purrr_0.3.4          readr_1.4.0          tidyr_1.1.3         
## [10] tibble_3.1.0         ggplot2_3.3.3        tidyverse_1.3.0     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.6          lubridate_1.7.9.2   png_0.1-7          
##  [4] assertthat_0.2.1    digest_0.6.27       utf8_1.2.1         
##  [7] R6_2.5.0            cellranger_1.1.0    backports_1.2.1    
## [10] reprex_0.3.0        evaluate_0.14       httr_1.4.2         
## [13] highr_0.8           pillar_1.6.0        GlobalOptions_0.1.2
## [16] rlang_0.4.11        readxl_1.3.1.9000   rstudioapi_0.13    
## [19] jquerylib_0.1.3     GetoptLong_1.0.4    rmarkdown_2.8      
## [22] munsell_0.5.0       broom_0.7.3         compiler_4.0.2     
## [25] modelr_0.1.8        xfun_0.22           pkgconfig_2.0.3    
## [28] shape_1.4.5         htmltools_0.5.1.1   tidyselect_1.1.0   
## [31] fansi_0.4.2         crayon_1.4.1        dbplyr_2.0.0       
## [34] withr_2.4.1         jsonlite_1.7.2      gtable_0.3.0       
## [37] lifecycle_1.0.0     DBI_1.1.0           magrittr_2.0.1     
## [40] scales_1.1.1        cli_2.5.0           stringi_1.5.3      
## [43] farver_2.1.0        fs_1.5.0            xml2_1.3.2         
## [46] bslib_0.2.4         ellipsis_0.3.2      generics_0.1.0     
## [49] vctrs_0.3.8         rjson_0.2.20        RColorBrewer_1.1-2 
## [52] tools_4.0.2         glue_1.4.2          hms_1.0.0          
## [55] parallel_4.0.2      yaml_2.2.1          clue_0.3-58        
## [58] colorspace_2.0-0    cluster_2.1.0       rvest_0.3.6        
## [61] knitr_1.33          haven_2.3.1         sass_0.3.1
```
